avocado/core/varianter.py

```diff
@@ -45,9 +45,29 @@ def generate_variant_id(variant):
     :return: String compounded of ordered node names and a hash of all
              values.
     """
+    def variantname(variant):
+        """
+        : param variant: Avocado test variant (list of TreeNode-like objects)
+        : return: Complete variant name string
+        """
+        full_name = []
+        for node in variant:
```

Comment:
I will wait for your comment on my first comment here, because looks like the logic will be changed.